src/net.h

```diff
@@ -1096,7 +1093,7 @@ class CConnman
     std::vector<ListenSocket> vhListenSocket;
     std::atomic<bool> fNetworkActive{true};
     bool fAddressesInitialized{false};
-    CAddrMan addrman;
+    CAddrMan& addrman;
```

Comment:
shared_ptr implies shared ownership of a resource. In this case, CConnman and PeerManager don't own the addrman, they just hold a reference to it.

We already use this same model for holding references to mempool, chainman, conman, etc, where the resource is owned by NodeContext.

Reply:
> shared_ptr implies shared ownership of a resource. In this case, CConnman and PeerManager don't own the addrman, they just hold a reference to it.

`shared_ptr` would just destroy the object when its last instance is destroyed. The fact that we know that this is going to be the instance in `NodeContext` does not make it less useful.

The problem with the bare pointer (or bare reference) is that it can become dangling. `shared_ptr` aims to fix exactly that problem.

> We already use this same model for holding references to mempool, chainman, conman, etc, where the resource is owned by NodeContext.

Right, that can be improved with `shared_ptr` too.

Reply:
Another good reason to leave this as a reference is that it explicitly communicates that there *must* be a reference to an addrman, and that we can 'dereference' addrman throughout peerman without checking for nullness first.

In any case, I agree that this discussion can be deferred to another issue/PR.